### PR TITLE
add support for HLS_METADATA

### PR DESCRIPTION
**Pull request by xfreid**
Signed-off-by: xilinxfei <fei@xilinx.com>

#### Problem solved by the commit
Add support for HLS_METADATA

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
N/A

#### How problem was solved, alternative solutions (if any) and why they were rejected
N/A

#### Risks (if any) associated the changes in the commit
None

#### What has been tested and how, request additional testing if necessary
Manual and unit test

#### Documentation impact (if any)
None

### CODE REVIEW

File: src/runtime_src/core/include/xclbin.h [hunk 1]
```diff
@@ -176,6 +176,7 @@ extern "C" {
         OVERLAY                = 30,
```
Comment (rozumx) [inline src/runtime_src/core/include/xclbin.h:176]: Copyright date (line 2) should be updated.

File: src/runtime_src/tools/xclbinutil/CMakeLists.txt [hunk 2]
```diff
@@ -147,6 +147,9 @@ if(NOT WIN32)
   set(TEST_OPTIONS " --resource-dir ${CMAKE_CURRENT_SOURCE_DIR}/unittests/MCS")
```
Comment (rozumx) [inline src/runtime_src/tools/xclbinutil/CMakeLists.txt:147]: Copyright date (line 2) needs to be updated.

File: src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2018, 2022 Xilinx, Inc
```
Comment (rozumx) [inline src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx:2]: Copyright date needs to be updated.

File: src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+  sectionInfo->supportedAddFormats.push_back(FormatType::raw);
+
+  sectionInfo->supportedDumpFormats.push_back(FormatType::raw);
+
+  addSectionType(std::move(sectionInfo));
+}
+
+void
+SectionIPMetadata::marshalToJSON(char* _pDataSection,
```
Comment (rozumx) [inline src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx:41]: Going forward, we will no longer be using the underscore '_'prefixes for argument names.

File: src/runtime_src/tools/xclbinutil/SectionIPMetadata.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (C) 2018 - 2022 Xilinx, Inc
```
Comment (rozumx) [inline src/runtime_src/tools/xclbinutil/SectionIPMetadata.h:2]: Copyright date.

File: src/runtime_src/tools/xclbinutil/unittests/FixedKernel/FixedKernel.py [hunk 6]
```diff
@@ -126,7 +126,7 @@ def main():
   
   step = '''3) Read in fixed ps kernel, updated and validate the sections
    fixed ps kernel file: fixed_kernel_add_2.json
-   this file set the funcitonal to 1 (i.e. "PrePost")'''
+   this file set the functional to 1 (i.e. "PrePost")'''
```
Comment (rozumx) [inline src/runtime_src/tools/xclbinutil/unittests/FixedKernel/FixedKernel.py:129]: Thank you for correcting the spelling issue.

**Review by rozumx — APPROVED** [review]
Overall the submission looks good.  There is though issues with the copyright dates (they need to be updated).

File: src/runtime_src/tools/xclbinutil/SectionIPMetadata.h (hunk 5 above)
Comment (xfreid) [reply src/runtime_src/tools/xclbinutil/SectionIPMetadata.h]: copyright date updated

File: src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx (hunk 4 above)
Comment (xfreid) [reply src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx]: removed "_"prefixes

File: src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx (hunk 3 above)
Comment (xfreid) [reply src/runtime_src/tools/xclbinutil/SectionIPMetadata.cxx]: copyright date updated

File: src/runtime_src/tools/xclbinutil/CMakeLists.txt (hunk 2 above)
Comment (xfreid) [reply src/runtime_src/tools/xclbinutil/CMakeLists.txt:147]: copyright date updated

File: src/runtime_src/core/include/xclbin.h (hunk 1 above)
Comment (xfreid) [reply src/runtime_src/core/include/xclbin.h:176]: copyright date updated

**Pull request merged**